docs/src/docs/08-testing.md

```diff
@@ -0,0 +1,236 @@
+import currentProjectIdStore from './stores/currentProjectIdStore'
+import actions from './actions'
+import getters from './getters'
+
+reactor.registerStores({
+  currentProjectId: currentProjectIdStore,
+  projects: projectStore,
+})
+
+export default { getters, actions }
+```
+
+#### `stores/currentProjectIdStore.js`
+
+```javascript
+import { Store } from 'nuclear-js'
+import { CHANGE_CURRENT_PROJECT_ID } from '../actionTypes'
+
+export default Store({
+  getInitialState() {
+    return null
+  },
+
+  initialize() {
+    this.on(CHANGE_CURRENT_PROJECT_ID, (currentId, newId) => newId)
+  },
+})
+```
+
+#### `stores/projectStore.js`
+
+```javascript
+import { Store, toImmutable } from 'nuclear-js'
+import { LOAD_PROJECTS } from '../actionTypes'
+
+export default Store({
+  getInitialState() {
+    // will maintain a map of project id => project object
+    return toImmutable({})
+  },
+
+  initialize() {
+    this.on(LOAD_PROJECTS, loadProjects)
+  },
+})
+
+/**
+ * @param {Immutable.Map} state
+ * @param {Object} payload
+ * @param {Object[]} payload.data
+ * @return {Immutable.Map} state
+ */
+function loadProjects(state, payload) {
+  return state.withMutations(state => {
+    payload.forEach(function(project) {
+      state.set(project.id, project)
+    })
+  })
+}
+```
+
+#### `actions.js`
+
+```javascript
+import Api from '../utils/api'
+import reactor from '../reactor'
+import { LOAD_PROJECTS } from './actionTypes'
+
+export default {
+  fetchProjects() {
+    return Api.fetchProjects.then(projects => {
+      reactor.dispatch(LOAD_PROJECTS, {
+        data: projects,
+      })
+    })
+  },
+
+  /**
+   * @param {String} id
+   */
+  setCurrentProjectId(id) {
+    reactor.dispatch(CHANGE_CURRENT_PROJECT_ID, id)
+  },
+}
+```
+
+#### `getters.js`
+
+```
+cosnt projectsMap = ['projects']
+
+const currentProjectId = ['currentProjectId']
+
+const currentProject = [
+  currentProjectId,
+  projectsMap,
+  (id, projects) => projects.get(id)
+]
+
+export default { projectsMap, currentProject, currentProjectId }
+```
+
+### Tests
+
+Given our module we want to test the following:
+
+- Using `actions.setCurrentProjectId()` sets the correct id using the `currentProjectId` getter
+
+- When `Api.fetchProducts` is stubbed with mock data, calling `actions.fetchProjects` properly poulates
+  the projects store by using the `projectsMap` getter
+
+- When projects have been loaded and currentProjectId set, `currentProject` getter works
+
+**Testing Tools**
+
+We will use the following tools: mocha, sinon, expect.js.  The same testing ideas can be implemented with a variety of tools.
+
+#### `tests.js`
+
+```javascript
+import reactor from '../reactor'
+import Api from '../utils/api'
+import expect from 'expect'
+
+// module under test
+import Project from './index'
+
+let mockProjects = [
+  { id: '123-abc', name: 'project 1' },
+  { id: '456-cdf', name: 'project 2' },
+]
+
+describe('modules/Project', () => {
+  afterEach(() => {
+    flux.reset()
```

Comment:
Good catch @balloob - `flux` is actually a wrapper we use for NuclearJS at Optimizely. In the case of the example tests, I believe it should be referencing `reactor`, yes. I'll send a PR fixing this. Thanks. :+1: